src/libasr/pickle.h

```diff
@@ -17,8 +17,8 @@ namespace LCompilers {
     std::string pickle_tree(ASR::TranslationUnit_t &asr, bool colors, bool show_intrinsic_modules=false);
 
     // Print Json structure
-    std::string pickle_json(ASR::asr_t &asr, LocationManager &lm, bool show_intrinsic_modules=false);
-    std::string pickle_json(ASR::TranslationUnit_t &asr, LocationManager &lm, bool show_intrinsic_modules=false);
+    std::string pickle_json(ASR::asr_t &asr, LocationManager &lm, bool no_loc=false, bool show_intrinsic_modules=false);
+    std::string pickle_json(ASR::TranslationUnit_t &asr, LocationManager &lm, bool no_loc=false, bool show_intrinsic_modules=false);
```

Comment:
Let's not use default arguments, it leads to issues with overloads and clarify of what the values actually are. For example, if there is any code in LFortran that just called `pickle_json(asr, lm, true)`, then after this PR this very same code now will evaluate to `no_loc=true` and `show_intrinsic_modules=false`, which is not what was intended.

To fix this, do it like this:
```suggestion
    std::string pickle_json(ASR::TranslationUnit_t &asr, LocationManager &lm, bool no_loc, bool show_intrinsic_modules);
```
And then modify all call sites to supply all arguments explicitly.